src/node/internal/internal_utils.ts

```diff
@@ -49,11 +50,13 @@ export function slowCases(enc: string): Encoding | undefined {
     case 4:
       if (enc === 'UTF8') return UTF8;
       if (enc === 'ucs2' || enc === 'UCS2') return UTF16LE;
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-template-expression
       enc = `${enc}`.toLowerCase();
       if (enc === 'utf8') return UTF8;
       if (enc === 'ucs2') return UTF16LE;
       break;
     case 3:
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-template-expression
       if (enc === 'hex' || enc === 'HEX' || `${enc}`.toLowerCase() === 'hex') {
```

Comment:
`${enc}.toLowerCase() === 'hex'` would only ever be usefull if enc is not a string. 
Maybe the type of enc needs to be fixed on that function?